test/e2e/container_probe.go

```diff
@@ -41,9 +41,22 @@ var _ = Describe("Probing container", func() {
 	It("with readiness probe should not be ready before initial delay and never restart [Conformance]", func() {
 		p, err := podClient.Create(makePodSpec(probe.withInitialDelay().build(), nil))
 		expectNoError(err)
-		startTime := time.Now()
 
-		Expect(wait.Poll(poll, 90*time.Second, func() (bool, error) {
+		Expect(wait.Poll(poll, 120*time.Second, func() (bool, error) {
+			p, err := podClient.Get(p.Name)
+			if err != nil {
+				return false, err
+			}
+
+			if p.Status.Phase != api.PodRunning {
+				Logf("pod is not yet running; pod has phase %q.", p.Status.Phase)
+				return false, nil
+			}
+			return true, nil
+		})).NotTo(HaveOccurred(), "pod never became running")
+
+		startTime := time.Now()
```

Comment:
Ah, nevermind - I was thinking that the pod would be Running as soon as the container started, but since there are 2 containers that is not the case. I'm still worried this will end up flaky.